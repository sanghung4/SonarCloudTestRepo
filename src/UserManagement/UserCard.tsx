import React from 'react';

import {
  Box,
  Button,
  Grid,
  Link,
  Typography
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { ChevronRightIcon } from 'icons';
import { CoercedUser } from 'UserManagement';

type Props = {
  user: CoercedUser | any;
  onRowClick: (row: CoercedUser) => void;
  rejectedUser?: boolean;
  index: number;
  viewUserId: string;
};

function UserCard(props: Props) {
  /**
   * Custom hooks
   */
  const { t } = useTranslation();

  return (
    <Box
      borderBottom={1}
      py={2}
      width={1}
      sx={{
        borderColor: 'secondary03.main'
      }}
    >
      <Grid
        container
        alignItems="flex-end"
        data-testid={`row_${props?.user.email}`}
      >
        <Grid item>
          <Box pb={1}>
            <Typography
              sx={{
                fontWeight: 500,
                color: 'primary.main'
              }}
            >
              {t('common.emailAddress')}
            </Typography>
            <Link
              href={`mailto:${props.user.email}`}
              sx={{ color: 'primary02.main' }}
              data-testid={`emailAddress-${props?.index}`}
            >
              {props.user.email}
            </Link>
          </Box>
          <Typography
            sx={{
              fontWeight: 500,
              color: 'primary.main'
            }}
          >
            {t('common.name')}
          </Typography>
          <div
            style={{ display: 'inline' }}
            data-testid={`firstName-${props?.index}`}
          >
            {` ${props.user.firstName} `}
          </div>
          <div
            style={{ display: 'inline' }}
            data-testid={`lastName-${props?.index}`}
          >
            {`${props.user.lastName}`}
          </div>
        </Grid>
        <Grid item display="flex" justifyContent="flex-end" flexGrow={1}>
          <Button
            onClick={() => props?.onRowClick(props.user)}
            endIcon={<ChevronRightIcon />}
            variant="text"
            sx={{ p: 0 }}
            data-testid={`viewUser-${props?.viewUserId
              .toLowerCase()
              .replaceAll(' ', '-')}-${props?.index}`}
          >
            {t('user.viewUser')}
          </Button>
        </Grid>
      </Grid>
    </Box>
  );
}

export default UserCard;
