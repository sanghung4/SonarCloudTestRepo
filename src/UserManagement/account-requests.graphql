query unapprovedAccountRequests($accountId: String!) {
  unapprovedAccountRequests(accountId: $accountId) {
    id
    email
    firstName
    lastName
    companyName
    phoneNumber
    roleId
    phoneType 
    createdAt
  }
}

query getApprovedAccountRequests($accountId: String) {
  accountUsers(accountId: $accountId) {
    id
    email
    firstName
    lastName
    phoneNumber
    contactUpdatedAt
    contactUpdatedBy
    role {
      id
      name
    }
    approverId
    phoneType 
  }
}

query rejectedAccountRequests($accountId: String!) {
  rejectedAccountRequests(accountId: $accountId) {
    id
    email
    firstName
    lastName
    companyName
    phoneNumber
    phoneType
    rejectionReason
    rejectedAt
    rejectedBy
  }
}
