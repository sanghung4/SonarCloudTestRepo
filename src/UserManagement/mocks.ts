import {
  UnapprovedAccountRequestsDocument,
  GetApprovedAccountRequestsDocument,
  RejectedAccountRequestsDocument
} from 'generated/graphql';
import { GraphQLError } from 'graphql';

export const mocks = [
  {
    request: {
      query: UnapprovedAccountRequestsDocument,
      variables: {
        accountId: '123456'
      }
    },
    result: {
      data: {
        unapprovedAccountRequests: [
          {
            id: 'f355e04d-9be3-45fd-82ea-3dab3707a883',
            email: 'tim+userman2@dialexa.com',
            firstName: 'John',
            lastName: 'Doe',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            roleId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            createdAt: '5/12/2021'
          },
          {
            id: 'ab4e4f52-4ea9-4a19-950c-4404b7c4bb2d',
            email: 'tim+userman3@dialexa.com',
            firstName: 'Elliot',
            lastName: 'Clark',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            roleId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            createdAt: '5/12/2021'
          },
          {
            id: '8410507e-afd9-48c2-8aae-d2dd88140aba',
            email: 'tim+userman5@dialexa.com',
            firstName: 'Brian',
            lastName: 'Bryan',
            companyName: '',
            phoneNumber: '(934) 024-8920',
            roleId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            createdAt: '5/12/2021'
          }
        ]
      }
    }
  },
  {
    request: {
      query: GetApprovedAccountRequestsDocument,
      variables: {
        accountId: '123456'
      }
    },
    result: {
      data: {
        accountUsers: [
          {
            id: '21766126-9cec-4d15-9b78-d2231da91931',
            email: 'eclipseadmindev+35648@morsco.com',
            firstName: 'john',
            lastName: 'admindev',
            phoneNumber: '(324) 234-2342',
            role: {
              id: 'c1ab78f0-d877-4fe8-bfef-99586d8e4131',
              name: 'Account Admin'
            },
            approverId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            contactUpdatedAt: '1/1/2021',
            contactUpdatedBy: '1/1/2021'
          },
          {
            id: '3286e61a-7b81-4cb3-8986-95e648b9080a',
            email: 'eclipsestanduserdev+35648@morsco.com',
            firstName: 'eclipse',
            lastName: 'standuserdev',
            phoneNumber: '(342) 423-4234',
            role: {
              id: '2419358e-5c74-4d07-a8d2-37c324afbe6c',
              name: 'Standard Access'
            },
            approverId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            contactUpdatedAt: '1/1/2021',
            contactUpdatedBy: '1/1/2021'
          },
          {
            id: '8410507e-afd9-48c2-8aae-d2dd88140aba',
            email: 'tim+userman6@dialexa.com',
            firstName: 'tim',
            lastName: 'standuserdev',
            phoneNumber: '(342) 423-4234',
            role: {
              id: '2419358e-5c74-4d07-a8d2-37c324afbe6c',
              name: 'Purchase with Approval'
            },
            approverId: null,
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            contactUpdatedAt: '1/1/2021',
            contactUpdatedBy: '1/1/2021'
          }
        ]
      }
    }
  },
  {
    request: {
      query: RejectedAccountRequestsDocument,
      variables: {
        accountId: '123456'
      }
    },
    result: {
      data: {
        rejectedAccountRequests: [
          {
            id: '4185a419-79cb-44de-9d94-1e29ce00bbbd',
            email: 'tim+userman1@dialexa.com',
            firstName: 'Tim',
            lastName: 'Waite',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            rejectionReason: {
              id: '5d506bb6-26b3-4b41-a670-5bd486eb0494',
              type: 'NOT_AUTHORIZED',
              description: 'Not authorized to be on E-Commerce'
            },
            rejectedAt: '5/12/2021',
            rejectedBy: 'seth+rddemo@dialexa.com',
            contactUpdatedAt: '1/2/2021'
          },
          {
            id: '4185a419-79cb-44de-9d94-1e29ce00bbbe',
            email: 'bob@dialexa.com',
            firstName: 'bob',
            lastName: 'bob',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            rejectionReason: {
              id: '5d506bb6-26b3-4b41-a670-5bd486eb0494',
              type: 'OTHER',
              description: 'Other'
            },
            rejectedAt: '5/12/2021',
            rejectedBy: 'seth+rddemo@dialexa.com',
            contactUpdatedAt: '1/2/2021'
          },
          {
            id: '4185a419-79cb-44de-9d94-1e29ce00bbbf',
            email: 'john@dialexa.com',
            firstName: 'john',
            lastName: 'john',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            rejectionReason: {
              id: '5d506bb6-26b3-4b41-a670-5bd486eb0494',
              type: 'NOT_A_COMPANY_MEMBER',
              description: 'Not a member of the company'
            },
            rejectedAt: '5/12/2021',
            rejectedBy: 'seth+rddemo@dialexa.com',
            contactUpdatedAt: '1/2/2021'
          },
          {
            id: '4185a419-79cb-44de-9d94-1e29ce00bbbg',
            email: 'Cameron@dialexa.com',
            firstName: 'Cameron',
            lastName: 'Cameron',
            companyName: '',
            phoneNumber: '(222) 222-2222',
            phoneType: {
              id: 'bcdb73c5-9d45-4933-b036-b5cf869a75ba',
              name: 'MOBILE'
            },
            rejectionReason: {
              id: '5d506bb6-26b3-4b41-a670-5bd486eb0494',
              type: 'NOT_A_COMPANY_MEMBER',
              description: 'Not a member of the company'
            },
            rejectedAt: '5/12/2021',
            rejectedBy: 'seth+rddemo@dialexa.com',
            contactUpdatedAt: '1/2/2021'
          }
        ]
      }
    }
  }
] as const;

export const mockOtherError = [
  {
    request: {
      query: UnapprovedAccountRequestsDocument,
      variables: {
        accountId: '123456'
      }
    },
    error: new GraphQLError('Other Error!')
  }
];

export const mockAccountNotFoundError = [
  {
    request: {
      query: UnapprovedAccountRequestsDocument,
      variables: {
        accountId: '123456'
      }
    },
    error: new GraphQLError('{"error":"Account with given ID not found."}')
  }
];
