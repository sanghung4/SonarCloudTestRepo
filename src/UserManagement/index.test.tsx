import React from 'react';

import { render } from 'test-utils/TestWrapper';
import { fireEvent, waitFor } from '@testing-library/react';

import UserManagement from 'UserManagement';
import {
  mocks,
  mockAccountNotFoundError,
  mockOtherError
} from 'UserManagement/mocks';
import { mocks as homeMocks } from 'Home/tests/mocks';
import { SelectedAccounts } from 'providers/SelectedAccountsProvider';
import { mockHistory } from 'test-utils/mockRouter';

const selectedAccountsMock: SelectedAccounts = {
  billTo: {
    id: '123456'
  }
};

const mockFns = {
  history: { ...mockHistory },
  pushAlert: jest.fn()
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => mockFns.history
}));

jest.mock('@dialexa/reece-component-library', () => ({
  ...jest.requireActual('@dialexa/reece-component-library'),
  useSnackbar: () => ({ pushAlert: mockFns.pushAlert })
}));

jest.mock('date-fns', () => ({
  ...jest.requireActual('date-fns'),
  differenceInDays: () => 3
}));

describe('User management tests', () => {
  it('User Management render matches the snapshot', () => {
    const { container } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });

    expect(container).toMatchSnapshot();
  });

  it('"User Management - Waiting for Approval" data is displayed correctly', async () => {
    const { container } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const check = mocks[0].result.data.unapprovedAccountRequests;

    expect(rows.length).toBe(check.length + 1); // plus the top header row
    expect(rows[1]).toHaveTextContent(check[0].email);
    expect(rows[2]).toHaveTextContent(check[1].email);
    expect(rows[3]).toHaveTextContent(check[2].email);
  });

  it('"User Management - Approved" data is displayed correctly', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const approvedTab = getByTestId('tab-Approved');
    fireEvent.click(approvedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const check = mocks[1].result.data.accountUsers;

    expect(rows.length).toBe(check.length + 1); // plus the top header row
    expect(rows[1]).toHaveTextContent(check[2].email);
    expect(rows[2]).toHaveTextContent(check[1].email);
  });

  it('"User Management - Rejected" data is displayed correctly', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const rejectedTab = getByTestId('tab-Rejected');
    fireEvent.click(rejectedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const check = mocks[2].result.data.rejectedAccountRequests;

    expect(rows.length).toBe(check.length + 1); // plus the top header row
    expect(rows[1]).toHaveTextContent(check[3].email);
    expect(rows[2]).toHaveTextContent(check[2].email);
    expect(rows[3]).toHaveTextContent(check[1].email);
    expect(rows[4]).toHaveTextContent(check[0].email);
  });

  it('Sort first name in ascending order on the "Rejected" table', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[2].result.data.rejectedAccountRequests;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const rejectedTab = getByTestId('tab-Rejected');
    fireEvent.click(rejectedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const firstNameColumn = rows[0].getElementsByTagName('th')[1];

    fireEvent.click(firstNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[1].firstName);
    expect(rows[2]).toHaveTextContent(check[3].firstName);
    expect(rows[3]).toHaveTextContent(check[2].firstName);
    expect(rows[4]).toHaveTextContent(check[0].firstName);
  });

  it('Sort first name in descending order on the "Rejected" table', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[2].result.data.rejectedAccountRequests;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const rejectedTab = getByTestId('tab-Rejected');
    fireEvent.click(rejectedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const firstNameColumn = rows[0].getElementsByTagName('th')[1];

    fireEvent.click(firstNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    fireEvent.click(firstNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[0].firstName);
    expect(rows[2]).toHaveTextContent(check[2].firstName);
    expect(rows[3]).toHaveTextContent(check[3].firstName);
    expect(rows[4]).toHaveTextContent(check[1].firstName);
  });

  it('Sort last name in ascending order on the "Rejected" table', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[2].result.data.rejectedAccountRequests;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const rejectedTab = getByTestId('tab-Rejected');
    fireEvent.click(rejectedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const lastNameColumn = rows[0].getElementsByTagName('th')[2];
    fireEvent.click(lastNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[1].lastName);
    expect(rows[2]).toHaveTextContent(check[3].lastName);
    expect(rows[3]).toHaveTextContent(check[2].lastName);
    expect(rows[4]).toHaveTextContent(check[0].lastName);
  });

  it('Sort last name in descending order on the "Rejected" table', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[2].result.data.rejectedAccountRequests;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const rejectedTab = getByTestId('tab-Rejected');
    fireEvent.click(rejectedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const lastNameColumn = rows[0].getElementsByTagName('th')[2];
    fireEvent.click(lastNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    fireEvent.click(lastNameColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[0].lastName);
    expect(rows[2]).toHaveTextContent(check[2].lastName);
    expect(rows[3]).toHaveTextContent(check[3].lastName);
    expect(rows[4]).toHaveTextContent(check[1].lastName);
  });

  it('show the Approved table sorted by Role in ascending order', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[1].result.data.accountUsers;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const approvedTab = getByTestId('tab-Approved');
    fireEvent.click(approvedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const roleColumn = rows[0].getElementsByTagName('th')[3];

    fireEvent.click(roleColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[0].role.name);
    expect(rows[2]).toHaveTextContent(check[2].role.name);
    expect(rows[3]).toHaveTextContent(check[1].role.name);
  });

  it('show the Approved table sorted by Role in descending order', async () => {
    const { container, getByTestId } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks
    });
    const check = mocks[1].result.data.accountUsers;

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    const approvedTab = getByTestId('tab-Approved');
    fireEvent.click(approvedTab);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    const rows = container.getElementsByTagName('tr');
    const roleColumn = rows[0].getElementsByTagName('th')[3];

    fireEvent.click(roleColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    fireEvent.click(roleColumn);
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(rows[1]).toHaveTextContent(check[1].role.name);
    expect(rows[2]).toHaveTextContent(check[2].role.name);
    expect(rows[3]).toHaveTextContent(check[0].role.name);
  });

  it('User Management render when account not found error in response for employee', async () => {
    const clearAccounts = jest.fn();
    const { container } = render(<UserManagement />, {
      selectedAccountsConfig: {
        selectedAccounts: selectedAccountsMock,
        clearAccounts: clearAccounts
      },
      authConfig: homeMocks.authorized,
      mocks: mockAccountNotFoundError
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(clearAccounts).toBeCalled();
    expect(mockFns.history.push).toBeCalledWith('/select-accounts');
  });

  it('User Management render when other error in response for employee', async () => {
    const clearAccounts = jest.fn();
    const { container } = render(<UserManagement />, {
      selectedAccountsConfig: {
        selectedAccounts: selectedAccountsMock,
        clearAccounts: clearAccounts
      },
      authConfig: homeMocks.authorized,
      mocks: mockOtherError
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockFns.pushAlert).toBeCalled();
  });

  it('User Management render when found an error in response for non employee', async () => {
    const { container } = render(<UserManagement />, {
      selectedAccountsConfig: { selectedAccounts: selectedAccountsMock },
      mocks: mockOtherError
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(mockFns.pushAlert).toBeCalled();
  });
});
