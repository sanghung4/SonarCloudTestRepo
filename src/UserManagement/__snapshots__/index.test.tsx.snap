// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User management tests User Management render matches the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1kbjjd5-MuiContainer-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1yp6lty-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-n1sla7-MuiBreadcrumbs-separator"
        >
          <svg>
            chevron-right.svg
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
            data-testid="user-management-breadcrumb"
          >
            User Management
          </span>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-1obe4va-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10 css-1s6wx5p-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
            data-testid="user-management-title"
          >
            User Management
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2 css-1qspkho-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-wyox7y-MuiButtonBase-root-MuiButton-root"
            data-testid="invite-new-user-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
            >
              <svg>
                add.svg
              </svg>
            </span>
            Invite new user
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gj3ka-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiTabs-root css-jlxbv8-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-kqhgxw-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="Selected Users"
            class="MuiTabs-flexContainer css-1pyy021-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1sakcj1-MuiButtonBase-root-MuiTab-root"
              data-testid="tab-Waiting for Approval"
              role="tab"
              tabindex="0"
              type="button"
            >
              Waiting for Approval
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sakcj1-MuiButtonBase-root-MuiTab-root"
              data-testid="tab-Approved"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Approved
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sakcj1-MuiButtonBase-root-MuiTab-root"
              data-testid="tab-Rejected"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Rejected
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-18hhhr3-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="MuiTableContainer-root css-8cb3ox-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-qzyhip-MuiTable-root"
          data-testid="table"
          role="table"
          style="min-width: 0px;"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-bqkitb-MuiTableRow-root"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Email Address
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 65 0 auto; min-width: 0px; width: 65px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      First Name
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 65 0 auto; min-width: 0px; width: 65px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Last Name
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 65 0 auto; min-width: 0px; width: 65px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Role
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-l2k3ri-MuiTableCell-root"
                colspan="1"
                role="columnheader"
                scope="col"
                style="box-sizing: border-box; flex: 65 0 auto; min-width: 0px; width: 65px; cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-11d6wci-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-auto css-289t-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-13o7eu2"
                    >
                      Day(s) Pending
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1c16yah-MuiGrid-root"
                  >
                    <svg
                      class="MuiBox-root css-1isemmb"
                      data-testid="createdAt-descending"
                      sx="[object Object]"
                    >
                      sorting-desc.svg
                    </svg>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                colspan="6"
              >
                <div
                  class="MuiBox-root css-18glhic"
                >
                  <div
                    class="MuiBox-root css-hyxlzm"
                  >
                    <span
                      aria-valuenow="100"
                      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 50px; height: 50px; transform: rotate(-90deg);"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="19.5"
                          stroke-width="5"
                          style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 50px; height: 50px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="19.5"
                          stroke-width="5"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-11ezxrp"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1vbz8i-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-auto css-kqwn4m-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-19z5v8g"
              data-testid="total-result(s)-count"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
